examples/example-gateway/clients/baz/baz.go

```diff
@@ -66,11 +68,21 @@ func NewClient(config *zanzibar.StaticConfig, gateway *zanzibar.Gateway) *BazCli
 type SimpleServiceClient struct {
 	thriftService string
 	client        zt.TChanClient
+	// timeout is the default timeout if upstream does not have one
+	timeout time.Duration
 }
 
 // Call ...
 func (c *SimpleServiceClient) Call(ctx context.Context, reqHeaders map[string]string, args *baz.SimpleService_Call_Args) (map[string]string, *baz.BazResponse, error) {
 	var result baz.SimpleService_Call_Result
+
+	_, ok := ctx.Deadline()
```

Comment:
This looks like a timeout check.

Every `ctx` should have a `Deadline()` I think it's best if we move this logic here ( https://github.com/uber/zanzibar/blob/03e905277787c3ec2670e12cd000d4d7878fbd18/runtime/router.go#L166 ).

When we have a new incoming request to the server. we should have a default timeout for the entire flow.

Now as for tchannel itself,  we want to allocate a `ContextBuilder` ( https://godoc.org/github.com/uber/tchannel-go#ContextBuilder ).

We want to set timeout and timeoutPerAttempt as well as the ParentContext.

So even if the ctx has a known Deadline() we still want to set a timeout on the context builder.
We will have to write a test for timeout propagation ( aka if ParentContext Deadline() is 2s and you set timeout to 5s does it pick 5s or 2s ).